src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVMKernels.hpp

```diff
@@ -37,6 +38,81 @@ namespace geosx
 namespace singlePhaseHybridFVMKernels
 {
 
+/******************************** PressureGradientKernel ********************************/
+
+struct PressureGradientKernel
+{
+  GEOSX_FORCE_INLINE
+  static void
+  compute( localIndex elemIndex,
+           localIndex numFacesInElem,
+           arrayView2d< real64 const > const faceCenter,
+           real64 const elemCenter[3],
+           arraySlice1d< localIndex const > const & elemToFaces,
+           arrayView1d< real64 const > const facePressure,
+           real64 const pres,
+           arrayView2d< real64 > const & presGradient )
+  {
+    array2d< real64 > coordinates( numFacesInElem+1, 4 ); 
+    array1d< real64 > pressures( numFacesInElem + 1 ); 
+    array1d< real64 > presGradientLocal( 4 ); 
```

Comment:
If you want, you can template this function on the number of faces in a cell and use the dispatch called `kernelLaunchSelectorFaceSwitch` defined [here](https://github.com/GEOSX/GEOSX/blob/d51a0af2c9d631d837efd71dd02bdc11403843da/src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVMKernels.hpp#L48). With that you will be able to replace the `array2d` with standard C-arrays.